src/lib/sap_ha/rpc_server.rb

```diff
@@ -82,40 +79,39 @@ def init_logger
       @fh.flock(File::LOCK_EX)
       @fh.sync = true
       @fh.flock(File::LOCK_UN)
-      @logger = Logger.new('/tmp/rpc_serv')
+      @logger = Logger.new("/tmp/rpc_serv")
```

Comment:
Please do not use a fixed path in a world writable directory. Any user can create a symlink pointing to a critical location and break the system or gain privileges.

E.g. if any user creates a `/tmp/rpc_serv` -> `/boot/vmlinuz` symlink then the system will not boot later as the kernel will be overwritten by some log messages.

- Use a proper function for creating unique temporary files ([Tempfile](https://ruby-doc.org/stdlib-2.5.3/libdoc/tempfile/rdoc/Tempfile.html) in Ruby)
- Or use a different directory (not world-writable), in this case `/var/log` looks like a good option

Reply:
Thanks. I’ll fix it when I’m back from vacation.

Reply:
This should be fixed for all versions.